### ISSUE TITLE
Does not notice unused assignment

### ISSUE THREAD

**Issue by aviv-ebates** [PyCQA/pylint#2840]
### Steps to reproduce
Pylint this file:
```py
# pylint: disable=missing-docstring,blacklisted-name

def foo(bar):
    if bar > 4:
        baz = bar + 3
        return baz

    baz = bar - 1  # This line has no effect
    return bar
```

### Current behavior
No warnings.

### Expected behavior
Assignment in line 8 `baz = bar - 1` is meaningless - `baz` is never read (e.g. because of a typo in next line). Pycharm knows to mark it as unused.

### pylint --version output
```
pylint 2.3.1
astroid 2.2.5
Python 3.7.1 (default, Oct 22 2018, 11:21:55)
[GCC 8.2.0]
```

**Comment by PCManticore** [comment]
Thank you @aviv-ebates This makes sense.

**Comment by clavedeluna** [comment]
Not to be pedantic but I think this needs to raise `unused-variable`. In case that's not obvious to other readers.

**Comment by nickdrozd** [comment]
Shorter example:

```python
def f():
    x = 4
    x = 5
```
